grails-app/controllers/OntologyController.groovy

```diff
@@ -71,23 +79,42 @@ class OntologyController {
             }
 
     def showConceptDefinition = {
-        def conceptPath = i2b2HelperService.keyToPath(params.conceptKey)
-        def node = OntNode.get(conceptPath)
+        OntologyTerm term = conceptsResourceService.getByKey(params.conceptKey)
+        Study study = term.study
 
-        //Check for study by visual attributes
-        if (node.visualattributes.contains("S")) {
-            def accession = node.sourcesystemcd
-            def study = Experiment.findByAccession(accession?.toUpperCase())
-            if (study) {
-                def folder = FmFolderAssociation.findByObjectUid(study.uniqueId?.uniqueId)?.fmFolder
-                if (folder) {
-                    def amTagTemplate = amTagTemplateService.getTemplate(folder.uniqueId)
-                    List<AmTagItem> metaDataTagItems = amTagItemService.getDisplayItems(amTagTemplate?.id)
-                    render(template: 'showStudy', model: [folder: folder, bioDataObject: study, metaDataTagItems: metaDataTagItems])
-                }
+        Experiment experiment
+        FmFolder folder
+        if (study?.ontologyTerm == term) {
+            // is the top study term
+            experiment = Experiment.findByAccession(study.id.toUpperCase())
+            if (experiment) {
+                folder = FmFolderAssociation.findByObjectUid(experiment.uniqueId?.uniqueId)?.fmFolder
+            }
+        }
+
+        if (!folder) {
+            log.debug "Could not find folder association; will look for tags"
+
+            def tags = []
+            // this solution is suboptimcal. The best would be to check that
+            // the table is i2b2metadata.i2b2, but there is no API method
+            // exposing that
+            if (!term.key.startsWith('\\\\xtrials\\')) {
```

Comment:
`ConceptKey` is unfortunately private to `core-db` (arguably it should be moved to core-api and `OntologyTerm::getKey()` should return a `ConceptKey`, but it doesn't). I'm not sure you mean the logic should be moved to `OntologyTerm`, but I'd rather not change the API unless absolutely necessary.